src/tools/debugger/mhpdbgc.c

```diff
@@ -1511,54 +1588,59 @@ void mhp_cmd(const char * cmd)
 
 static void mhp_print_ldt(int argc, char * argv[])
 {
+  static char lastldt[32];
   static char buffer[0x10000];
-  unsigned int *lp, *lp_=(unsigned int *)dpmi_get_ldt_buffer();
-  unsigned int base_addr, limit;
-  int type, type2, i;
+  unsigned int *lp, *lp_;
+  unsigned int base_addr, limit, i, lines=0;
+  int type, type2;
   unsigned int seg;
-  int lines=0, cache_mismatch;
+  int cache_mismatch;
   enum{Abit=0x100};
 
-  if (get_ldt(buffer) < 0) {
-    mhp_printf("error getting ldt\n");
+  if (!check_for_stopped())
+    return;
+
+  if (argc > 2 && !getval_ui(argv[2], 0, &lines)) {
```

Comment:
But IMHO it makes absolutely no sense
to specify the amount of lines together with
the descriptor value to print, as you will
obviously print only one in this case.
I just think the changes around here are an
unnecessary complications w/o real use.

Reply:
I would agree if I knew for sure how many ldt entries it was possible to print. I looked at [Wikipedia GDT page](https://en.wikipedia.org/wiki/Global_Descriptor_Table) and it suggests 8192 entries. Since we are unlikely to print all in one go we might need some way of starting at an offset. Of course I'm happy to remove the lines arg as I am very unlikely to use it myself, but just conscious of not wrecking it for others.

Reply:
8192 is a maximum value.
DPMI progs rarely use more than a few
dozens (well, windows can use much more).
While printing ranges may be useful, printing
a single entry you are interested in, is much,
much more useful when you debug something.
And these are the two entirely different operations
(print range and  print particular entry).
So how about the following:
1. Leave the current logic for the first invocation
2. If user pressed Enter again (repeat last cmd),
then you print next 16 entries no matter how much
you printed on step 1.

?